lib/ecto/changeset.ex

```diff
@@ -106,17 +111,19 @@ defmodule Ecto.Changeset do
 
   def change(%Changeset{changes: changes} = changeset, new_changes)
       when is_map(new_changes) do
-    %{changeset | changes: get_changed(changeset.model, changes, new_changes)}
+    changeset = ensure_has_types(changeset)
```

Comment:
Can a changeset be created only with functions from this module (I know it can, but do we support it)?
If not you are right - we can make sure we always have types.